src/main/java/nextstep/sessions/domain/SessionStatus.java

```diff
@@ -1,13 +1,45 @@
 package nextstep.sessions.domain;
 
-public enum SessionStatus {
-    PREPARING("준비중"),
-    RECRUITING("모집중"),
-    END("종료");
+import nextstep.common.Period;
 
-    private final String status;
+public class SessionStatus {
 
-    SessionStatus(String status) {
-        this.status = status;
+    private SessionProgressStatus progressStatus;
+
+    private SessionRecruitmentStatus recruitmentStatus;
+
+    public SessionStatus(Period date) {
+        if (date.isPreparing()) {
+            progressStatus = SessionProgressStatus.PREPARING;
+            recruitmentStatus = SessionRecruitmentStatus.RECRUITING;
+            return;
+        }
+        if (date.isInProgress()) {
+            progressStatus = SessionProgressStatus.PROGRESSING;
+            recruitmentStatus = SessionRecruitmentStatus.RECRUITING;
+            return;
+        }
+        progressStatus = SessionProgressStatus.END;
+        recruitmentStatus = SessionRecruitmentStatus.NON_RECRUITMENT;
+    }
+
+    public SessionStatus(SessionProgressStatus progressStatus, SessionRecruitmentStatus recruitmentStatus) {
+        this.progressStatus = progressStatus;
+        this.recruitmentStatus = recruitmentStatus;
+    }
+
+    public SessionProgressStatus getProgressStatus() {
+        return progressStatus;
+    }
+
+    public SessionRecruitmentStatus getRecruitmentStatus() {
+        return recruitmentStatus;
+    }
+
+    public boolean isNotRecruiting() {
+        if (recruitmentStatus == SessionRecruitmentStatus.NON_RECRUITMENT) {
+            return true;
+        }
+        return false;
```

Comment:
```suggestion
        return !recruitmentStatus.isRecruiting();
```
enum도 객체이기 때문에 메시지를 보내는 방식으로 구현 가능하다.